File: backend/DormitoryManagement.Data/Models/Chair.cs
using DormitoryManagement.Data;
using System.ComponentModel.DataAnnotations;

namespace DormitoryManagement.Data.Models
{
    public class Chair
    {
        [Key]
        [MaxLength(50)]
        public int SerialNumber { get; set; } // Первинний ключ
        [MaxLength(50)]
        public int ConditionId { get; set; } // Зовнішній ключ
        public Condition Condition { get; set; } // Навігаційна властивість
        [MaxLength(50)]
        public int TypeId { get; set; } // Зовнішній ключ
        public ChairType Type { get; set; } // Навігаційна властивість
        [MaxLength(50)]
        public string RoomNumber { get; set; }
        public Room? Room { get; set; }
    }
}

File: frontend/src/app/models/Chair.ts
import {ChairType, Condition} from './Enums';

export interface Chair {
  serialNumber: number;
  condition: Condition;
  type: ChairType;
  roomNumber: string;
  expanded?: boolean;
}

export interface ChairDTO {
  serialNumber?: number; // Опціонально, бо не завжди потрібен при оновленні
  conditionId: number;
  typeId: number;
  roomNumber: string;
}
